### PR TITLE
Add BPF architecture target support

### PR DESCRIPTION
**Pull request by cupermir**
BPF is a virtual machine and associated ISA that resides in the Linux kernel.  Initially intended for user-level packet capture and filtering, BPF is nowadays generalized to serve as a general-purpose infrastructure also for non-networking purposes.

Signed-off-by: Cupertino Miranda <cupertino.miranda@oracle.com>

### CODE REVIEW

**Review by cpackham — APPROVED** [review]

File: config/arch/bpf.in [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# BPF specific configuration file
+
+## no-package
+## select ARCH_SUPPORTS_EITHER_ENDIAN
+## select ARCH_DEFAULT_LE
+## select GCC_VERY_NEW
+## select BINUTILS_VERY_NEW
```
Comment (cpackham) [inline config/arch/bpf.in:7]: It'd be good to replace these `VERY_NEW`s with actual versions as soon as possible. Going on previous releases we won't see GCC 13 until April/May. Binutils on the other hand seem to have a fairly regular release cadence.

Comment (cupermir) [reply config/arch/bpf.in:7]: For now BPF target only makes sense for master branches as it is still under development. Once time comes and if fixes are backported to earlier version I will make that change. ;-)

**Comment by cupermir** [comment]
Is there anything I should do for "1 workflow awaiting approval" to progress ?

**Comment by cpackham** [comment]
Nope that's on me. I was going to ask you to rebase and force push but I
can sort that out from my end.

**Comment by cupermir** [comment]
I have rebased it manually to make sure it merges properly.
Looking forward to the merge. 
Thanks :+1:

**Review by cpackham — APPROVED** [review]

**Pull request merged**